jwst/datamodels/wcs_ref_models.py

```diff
@@ -667,30 +667,46 @@ class FilteroffsetModel(ReferenceFileModel):
     schema_url = "http://stsci.edu/schemas/jwst_datamodel/filteroffset.schema"
     reftype = "filteroffset"
 
-    def __init__(self, init=None, filters=None, **kwargs):
+    def __init__(self, init=None, filters=None, instrument=None, **kwargs):
         super(FilteroffsetModel, self).__init__(init, **kwargs)
         if filters is not None:
             self.filters = filters
+            if instrument is None or instrument not in ("NIRCAM", "MIRI", "NIRISS"):
+                raise ValueError("Please specify the instrument")
+            self.meta.instrument.name = instrument
 
     def populate_meta(self):
-        self.meta.instrument.name = "MIRI"
-        self.meta.instrument.detector = "MIRIMAGE"
-        self.meta.instrument.pfilter = "F1130W|F1140C|F2300C|F2100W|F1800W|\
-        F1550C|F560W|F2550WR|FND|F2550W|F1500W|F1000W|F1065C|F770W|F1280W|"
-
-    def on_save(self, path=None):
         self.meta.reftype = self.reftype
+        if self.meta.instrument.name == "MIRI":
+            self.meta.instrument.detector = "MIRIMAGE"
+            self.meta.instrument.pfilter = "F1130W|F1140C|F2300C|F2100W|F1800W|\
+            F1550C|F560W|F2550WR|FND|F2550W|F1500W|F1000W|F1065C|F770W|F1280W|"
+        elif self.meta.instrument.name == "NIRCAM":
+            self.meta.instrument.pdetector = "NRCA1|NRCA2|NRCA3|NRCA4|NRCALONG|\
```

Comment:
Since the NIRCam ref files aren't being selected by detector, do we need this? Not even sure it'll work, because the `pdetector` attribute isn't in the schema. Or is this needed for something internally?

Reply:
Yes, this should be removed. It's left from one of the many iterations I did.